### PR TITLE
fix: check if stream_routes is disabled

### LINKED ISSUE #856: bug: when some features in APISIX are disabled, the shared context will cause some exceptions
* When the APISIX Ingress controller synchronizes data, it is expected that all resources can be successfully synchronized (in this issue, resources such as stream_routes will inevitably fail because stream_proxy is not configured, but APISIX Ingress congtroller does not handle this situation. It's a bug.

_Originally posted by @tao12345666333 in https://github.com/apache/apisix-ingress-controller/issues/851#issuecomment-1026944206_

### PR DESCRIPTION
**Pull request by lingsamuel**
Signed-off-by: Ling Samuel <lingsamuelgrace@gmail.com>

### Type of change:

- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

### What this PR does / why we need it:

Fixes #856 

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Comment by tao12345666333** [comment]
@lingsamuel Can we add an e2e test case to cover this scenario?

We need to create a new APISIX configuration file (disable stream_route) and start it.

**Comment by lingsamuel** [comment]
@tao12345666333 done

**Review by tao12345666333 — APPROVED** [review]
LGTM

**Comment by tao12345666333** [comment]
@lingsamuel #829 has been merged. You can try to update code base on master branch. Thanks

**Comment by tao12345666333** [comment]
ping @gxthrj @tokers  PTAL, thanks!

File: test/e2e/ingress/sanity.go [hunk 1]
```diff
@@ -186,3 +186,49 @@ var _ = ginkgo.Describe("leader election", func() {
...
+	opts := &scaffold.Options{
+		Name:                  "default",
+		Kubeconfig:            scaffold.GetKubeconfig(),
+		APISIXConfigPath:      "testdata/apisix-stream-disabled.yaml",
+		IngressAPISIXReplicas: 1,
+		HTTPBinServicePort:    80,
+		APISIXRouteVersion:    "apisix.apache.org/v2beta3",
+	}
+	s := scaffold.NewScaffold(opts)
+	ginkgo.It("/ip should return your ip", func() {
```
Comment (tokers) [inline test/e2e/ingress/sanity.go:200]: What's the correlation between this case and "stream route disabled"?

Comment (tao12345666333) [reply test/e2e/ingress/sanity.go:200]: @tokers please see #856  
Previously, if this stream_route was disabled, it did not work properly

Comment (tokers) [reply test/e2e/ingress/sanity.go:200]: OK. I suggest adding some comments to explain it as the case doesn't have any relations with the stream route.

**Review by gxthrj — APPROVED** [review]
LGTM

**Review by tokers — APPROVED** [review]

**Pull request merged**